parsl/dataflow/dflow.py

```diff
@@ -397,12 +397,14 @@ def handle_app_update(self, task_id, future):
 
         """
 
-        if not self.tasks[task_id]['app_fu'].done():
+        task_id = task_record['id']
+
+        if not task_record['app_fu'].done():
             logger.error("Internal consistency error: app_fu is not done for task {}".format(task_id))
-        if not self.tasks[task_id]['app_fu'] == future:
+        if not task_record['app_fu'] == future:
             logger.error("Internal consistency error: callback future is not the app_fu in task structure, for task {}".format(task_id))
 
-        self.memoizer.update_memo(task_id, self.tasks[task_id], future)
+        self.memoizer.update_memo(task_id, task_record, future)
```

Comment:
Are changes to update_memo in a separate PR?

Reply:
It will be. Doing this for checkpointing/memo is more complicated than the substitions in #2015 .. #2018. This set of PRs is the easy stuff to lay the groundwork for that.